apps/hash-graph/lib/graph/src/api/rest/account.rs

```diff
@@ -51,18 +73,34 @@ impl RoutedResource for AccountResource {
                 "/account_groups",
                 Router::new()
                     .route("/", post(create_account_group::<S, A>))
-                    .route(
-                        "/:account_group_id/owners/:account_id",
-                        post(add_account_group_owner::<A>).delete(remove_account_group_owner::<A>),
-                    )
-                    .route(
-                        "/:account_group_id/admins/:account_id",
-                        post(add_account_group_admin::<A>).delete(remove_account_group_admin::<A>),
-                    )
-                    .route(
-                        "/:account_group_id/members/:account_id",
-                        post(add_account_group_member::<A>)
-                            .delete(remove_account_group_member::<A>),
+                    .nest(
+                        "/:account_group_id",
+                        Router::new()
+                            .route(
+                                "/owners/:account_id",
+                                post(add_account_group_owner::<A>)
+                                    .delete(remove_account_group_owner::<A>),
+                            )
+                            .route(
+                                "/admins/:account_id",
+                                post(add_account_group_admin::<A>)
+                                    .delete(remove_account_group_admin::<A>),
+                            )
+                            .route(
+                                "/members/:account_id",
+                                post(add_account_group_member::<A>)
+                                    .delete(remove_account_group_member::<A>),
+                            )
+                            .nest(
+                                "/permissions",
+                                Router::new()
+                                    .route("/add_owner", get(can_add_group_owner::<A>))
+                                    .route("/remove_owner", get(can_remove_group_owner::<A>))
+                                    .route("/add_admin", get(can_add_group_admin::<A>))
+                                    .route("/remove_admin", get(can_remove_group_admin::<A>))
+                                    .route("/add_member", get(can_add_group_member::<A>))
+                                    .route("/remove_member", get(can_remove_group_member::<A>)),
```

Comment:
The idea is to pass a permission instead which can be checked but the types are not encoded correctly. I'm going to change this after the issue described in https://github.com/hashintel/hash/pull/3376 is solved.